prov/efa/src/rdm/rxr_pkt_type_misc.c

```diff
@@ -76,11 +95,21 @@ ssize_t rxr_pkt_init_handshake(struct rxr_ep *ep,
 	 * Always include connid at the end of a handshake packet.
 	 * If peer cannot make use of connid, the connid will be ignored.
 	 */
-	connid_hdr = (struct rxr_handshake_opt_connid_hdr *)(pkt_entry->wiredata + pkt_entry->pkt_size);
+	connid_hdr = (struct rxr_handshake_opt_connid_hdr *)rxr_pkt_end(pkt_entry);
 	connid_hdr->connid = rxr_ep_raw_addr(ep)->qkey;
 	handshake_hdr->flags |= RXR_PKT_CONNID_HDR;
 	pkt_entry->pkt_size += sizeof(struct rxr_handshake_opt_connid_hdr);
 
+	/**
+	 * Include the optional hid(host id) if it is non-zero
+	 */
+	if (ep->host_id) {
+		hid_hdr = (struct rxr_handshake_opt_hid_hdr *)rxr_pkt_end(pkt_entry);
+		hid_hdr->hid = ep->host_id;
+		handshake_hdr->flags |= RXR_PKT_HID_HDR;
+		pkt_entry->pkt_size += sizeof(struct rxr_handshake_opt_hid_hdr);
+	}
+
```

Comment:
I think all this logic would be simplified if we simply reported the host_id all the time.  Other side can interpret 0 as "not provided"

Reply:
The reason for this check is for API compatibility in case the peer is on an older version - we cannot assume that host id is available.